packages/mui-material/src/Autocomplete/Autocomplete.js

```diff
@@ -625,8 +625,9 @@ const Autocomplete = React.forwardRef(function Autocomplete(inProps, ref) {
   const renderGroup = renderGroupProp || defaultRenderGroup;
   const defaultRenderOption = (props2, option) => {
     // Need to clearly apply key because of https://github.com/vercel/next.js/issues/55642
+    const { key, ...otherProps } = props2;
     return (
-      <li {...props2} key={props2.key}>
+      <li key={key} {...otherProps}>
         {getOptionLabel(option)}
       </li>
```

Comment:
Revert #40968 because of https://github.com/mui/material-ui/issues/39833#issuecomment-2101675127.